src/house.cpp

```diff
@@ -484,13 +487,13 @@ void Door::readAttr(AttrTypes_t attr, OTB::iterator& first, const OTB::iterator&
 	Item::readAttr(attr, first, last);
 }
 
-void Door::setHouse(House* house)
+void Door::setHouse(std::shared_ptr<House> house)
 {
-	if (this->house) {
+	if (!this->house.expired()) {
 		return;
 	}
 
-	this->house = house;
+	this->house = std::move(house);
```

Comment:
```suggestion
	this->house = house;
```

And take `const&` as param

Reply:
Changed